hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-2.x-replication/src/test/java/com/google/cloud/bigtable/hbase2_x/replication/HbaseToCloudBigtableReplicationEndpointTest.java

```diff
@@ -425,4 +418,29 @@ public void testWriteFailureToBigtableDoesNotStallReplication()
         "Value mismatch", TestUtils.getValue(0), CellUtil.cloneValue(actualCells.get(1)));
     Assert.assertEquals(0, actualCells.get(1).getTimestamp());
   }
+
+  @Test
+  public void testMutationReplicationWithWALEntryFilter() throws IOException, InterruptedException {
+    Put put = new Put(FILTERED_ROW_ENTRY);
+    put.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_ENTRY);
+    put.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_ENTRY);
+    hbaseTable.put(put);
+
+    TestUtils.waitForReplication(
+        () -> {
+          //  replicate Entries will be zero
+          return TestReplicationEndpoint.replicatedEntries.get() >= 0;
```

Comment:
how does this work? The test will not wait at all since repicatedEntires will be empty at the start. So if you don't wait, how does it prove that the row was filtered, vs the replication has not caught up yet? 

I think you want to put a "normal" row after the filtered row, wait for the normal row to replicate and then validate that normal row is there but "filtered" row is absent.

Reply:
Sure. I will add a normal row after the filtered row and make sure the replication actually catches up.

Reply:
Done.